File: 프로그래머스/unrated/181909. 접미사 배열/접미사 배열.java
import java.util.Arrays;

class Solution {
    public String[] solution(String my_string) {
        String[] answer = new String[my_string.length()];
        // answer의 길이는 my_string의 길이와 동일하게 선언한다
       
        for(int i = 0; i < answer.length; i++) {
            answer[i] = my_string.substring(i);
        } // i가 증가하면서 i번째부터 마지막 문자까지 answer에 만들어진다

        Arrays.sort(answer);
        // answer를 정렬한다
        
        return answer;
    }
}

File: 프로그래머스/0/181909. 접미사 배열/접미사 배열.js
function solution(my_string) {
    let answer = [];

    for (let i = 0; i < my_string.length; i++) {
    // answer 배열을 my_string의 길이와 동일한 길이로 초기화한다    
        answer.push(my_string.substring(i));
        // i번째부터 끝까지의 부분 문자열을 answer 배열에 추가한다
    }
    answer.sort();
    // answer 배열을 정렬한다
    
    return answer;
}
